### PR TITLE
PR: Handle Sympy foreground color setting following a color scheme change

### LINKED ISSUE #3798: Invert colors for "Pretty-Printing" symbolic math in IPython's dark background
The pretty printed symbolic math is unreadable when the the IPython console background is set to dark.

To reproduce:
1. Set preferences -> IPython console->Display->Background color to dark
2. Set preferences -> IPython console->Advanced->Use symbolic math
3. Restart ipython console
4. Enter a numeric command into Ipython and execute

Expected: Visible output
Got: only readable output when selected with the mouse to invert colors

* Spyder Version: 3.0.2
* Python Version: 3.5.2 64bit
* Operating system: Linux (Plasma5)

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.2 (OK)
pandas >=0.13.1  :  0.19.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
psutil >=0.3     :  5.0.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Change foreground color of Sympy to match current color scheme

se

### Issue(s) Resolved

Fixes #3798 
Depends on spyder-ide/spyder-kernels#137

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 1]
```diff
@@ -186,8 +186,14 @@ def set_color_scheme(self, color_scheme, reset=True):
             self.reset(clear=True)
         if not dark_color:
             self.silent_execute("%colors linux")
```
Comment (goanpeca) [inline spyder/plugins/ipythonconsole/widgets/shell.py]: What does this line do?

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 2]
```diff
@@ -186,8 +186,14 @@ def set_color_scheme(self, color_scheme, reset=True):
...
+            self.call_kernel(
+                interrupt=True,
+                blocking=False).set_sympy_forecolor(background_color='dark')
         else:
             self.silent_execute("%colors lightbg")
```
Comment (goanpeca) [inline spyder/plugins/ipythonconsole/widgets/shell.py]: What does this line do?

File: spyder/plugins/ipythonconsole/widgets/shell.py (hunk 1 above)
Comment (dalthviz) [reply spyder/plugins/ipythonconsole/widgets/shell.py]: We use this to change the colors of the tracebacks

image

[Here the docs for further info](https://ipython.readthedocs.io/en/stable/config/details.html?#terminal-colors)

File: spyder/plugins/ipythonconsole/widgets/shell.py (hunk 2 above)
Comment (dalthviz) [reply spyder/plugins/ipythonconsole/widgets/shell.py]: Same as above :)

File: spyder/plugins/ipythonconsole/widgets/shell.py (hunk 1 above)
Comment (goanpeca) [reply spyder/plugins/ipythonconsole/widgets/shell.py]: @dalthviz could we add this comment to the code :-) ?

File: README.md [hunk 3]
```diff
@@ -241,7 +241,7 @@ a Python version equal or greater than 3.5 or 2.7.
 * **Chardet**: Character encoding auto-detection in Python.
 * **Numpydoc**: Used by Jedi to get function return types from Numpydocstrings.
 * **Cloudpickle**: Serialize variables in the IPython kernel to send to Spyder.
-* **spyder-kernels** 1.4.0+: Jupyter kernels for the Spyder console.
+* **spyder-kernels** 1.5.0+: Jupyter kernels for the Spyder console.
```
Comment (ccordoba12) [inline README.md]: Please revert this.

File: requirements/conda.txt [hunk 4]
```diff
@@ -19,6 +19,6 @@ qtawesome >=0.5.7
 qtconsole >=4.5.2
 qtpy >=1.5.0
 sphinx
-spyder-kernels >=1.4.0
+spyder-kernels >=1.5.0
```
Comment (ccordoba12) [inline requirements/conda.txt]: Revert this.

File: setup.py [hunk 5]
```diff
@@ -213,7 +213,7 @@ def run(self):
     'pyzmq',
     'chardet>=2.0.0',
     'numpydoc',
-    'spyder-kernels>=1.4.0,<1.5.0',
+    'spyder-kernels>=1.5.0',
```
Comment (ccordoba12) [inline setup.py]: Revert this.

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 6]
```diff
@@ -185,9 +185,16 @@ def set_color_scheme(self, color_scheme, reset=True):
         if reset:
             self.reset(clear=True)
         if not dark_color:
+            # Needed to change the colors of the tracebacks
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/shell.py]: Please change this to

    # Needed to change the colors of tracebacks

**Pull request merged**